knimin/tests/test_ag_new_kit.py

```diff
@@ -0,0 +1,129 @@
+        self.mock_login_admin()
+        response = self.get('/ag_new_kit/')
+        self.assertEqual(response.code, 200)
+        for project in db.getProjectNames():
+            self.assertIn("<option value='%s'>%s</option>" %
+                          (project, project), response.body)
+        self.assertIn("%i</span> unassigned barcodes" %
+                      len(db.get_unassigned_barcodes()), response.body)
+
+    def test_post(self):
+        self.mock_login_admin()
+        kits = [1, 2]
+        swabs = [2, 2]
+        tag = 'abc'
+
+        # check for correct results
+        response = self.post('/ag_new_kit/',
+                             {'tag': tag,
+                              'projects': ['PROJECT2', 'PROJECT5'],
+                              'swabs': swabs,
+                              'kits': kits,
+                              })
+        kitinfo = loads(response.body)
+        self.assertEqual(len(kitinfo['kitinfo']), sum(kits))
+        for k in kitinfo['kitinfo']:
+            self.assertIn(tag, k[0])
+        self.assertEqual(len(kitinfo['kitinfo'][0][-1]), swabs[0])
+        self.assertEqual(len(kitinfo['kitinfo'][1][-1]), swabs[1])
+        self.assertEqual(kitinfo['fields'],
+                         "kit_id,password,verification_code,barcodes")
+
+        # missing argument
+        response = self.post('/ag_new_kit/',
+                             {'projects': ['PROJECT2', 'PROJECT5'],
+                              'swabs': swabs,
+                              'kits': kits,
+                              })
+        self.assertEqual(response.code, 400)
+
+        # too long tag
+        response = self.post('/ag_new_kit/',
+                             {'tag': 'toolongtag',
+                              'projects': ['PROJECT2', 'PROJECT5'],
+                              'swabs': swabs,
+                              'kits': kits,
+                              })
+        self.assertEqual(response.code, 500)
+        self.assertIn("Tag must be 4 or less characters", response.body)
+
+        # check project existens
+        response = self.post('/ag_new_kit/',
+                             {'tag': 'abc',
+                              'projects': ['doesNotExist', 'PROJECT5'],
+                              'swabs': swabs,
+                              'kits': kits,
+                              })
+        self.assertEqual(response.code, 500)
+        self.assertIn("Project(s) given don\'t exist in database:",
+                      response.body)
+
+        # check for empty swabs list
+        response = self.post('/ag_new_kit/',
+                             {'tag': tag,
+                              'projects': ['PROJECT2', 'PROJECT5'],
+                              'swabs': [],
+                              'kits': kits,
+                              })
+        self.assertEqual(response.code, 500)
+        self.assertIn("SET assigned_on = NOW() WHERE barcode IN ()",
```

Comment:
Agree. But we do at the moment, because we are to lazy to capture this situation in the handler or the python connector to SQL :-/ Add this to our todo list.